feat: add oauth2 imageFile

newAM opened the pull request:
There is no GET API to compare previous image to desired, this uploads the image each time.

oddlama commented on src/client.rs:
```diff
@@ -422,6 +422,46 @@ impl KanidmClient {
+
+        let path = Path::new(image_file);
+
+        let mine_str = match path.extension().and_then(|ext| ext.to_str()) {
```
```suggestion
        let mime_str = match path.extension().and_then(|ext| ext.to_str()) {
```

oddlama commented on src/client.rs:
```diff
@@ -422,6 +422,46 @@ impl KanidmClient {
+
+        let file_data = multipart::Part::bytes(image_data)
+            .file_name(filename)
+            .mime_str(mine_str)
```
```suggestion
            .mime_str(mime_str)
```

oddlama commented on src/client.rs:
```diff
@@ -422,6 +422,46 @@ impl KanidmClient {
+
+        let path = Path::new(image_file);
+
+        let mine_str = match path.extension().and_then(|ext| ext.to_str()) {
```
Can we add a `.to_lowercase()` here? I'd like to match case-insensitively on the extension.

newAM replied:
Yup, that's a good call, done!

newAM commented:
> I guess we can't do anything about reuploading at this point in time without tracking when we updated the image the last time somewhere. 

I think in most cases kanidm-provision will be run on the same server as kanidm, and even then it's not a lot of data, especially with SVGs.  Uploading each time isn't ideal, but I figure it's fine for now.

> Let's maybe just add a note about this to the description in the example json, so that users know about this.

I added the note to the example JSON and NixOS module!

oddlama commented:
LGTM!
